File: python/floodFillStack.py
import sys
import API
import cell

MAXSIZE = 512

class floodFillStack:
    def __init__(self):
        self.top = -1
        for i in range(0,512):
            self.memory = [cell.Cell()]

    def push(self, currentCell):
        if self.top >= MAXSIZE - 1:
            sys.stderr.write("Stack Overflow \n")
            return False
        else:
            self.top += 1
            self.memory[self.top] = currentCell
            return True

    def pop(self):
        if self.top < 0:
            sys.stderr.write("Stack Underflow \n")
            return
        else:
            popCell = self.memory[self.top]
            self.top -= 1
            return popCell
        return

    def isEmpty(self):
        if self.top < 0:
            return True
        else:
            return False

File: simulator/FloodFillStack.js
import { setMaxListeners } from "cluster";

function FloodFillStack() {
    this.memory = []
    this.top = -1
    for (var i = 0; i < 512; i++) {
        this.memory.push([0, 0, 0])
    }

    this.push = ([row, collumn, value]) => {
        if (self.top >= 512 - 1) {
            return false
        } else {
            self.top++
            self.memory[self.top] = [row, collumn, value]
            return true
        }
    }

    this.pop = () => {
        if (self.top < 0) {
            return
        } else {
            self.top--
            return self.memory[self.top + 1]
        }
    }

    this.isEmpty = () => {
        if (self.top < 0) {
            return true
        } else {
            return false
        }
    }

}